libevmasm/SemanticInformation.cpp

```diff
@@ -121,7 +121,9 @@ vector<SemanticInformation::Operation> SemanticInformation::readWriteOperations(
 			Location::Memory,
 			Effect::Write,
 			paramCount - 2,
-			paramCount - 1,
+			// Length is in paramCount - 1, but it is only a max length,
+			// there is no guarantee that the full area is written to.
+			{},
```

Comment:
Is this really the only case? What about ``CODECOPY`` and ``EXTCODECOPY``? Do they write zeroes beyond the respective code size or stop?

Reply:
(The yellowpaper reads like they stop, but I haven't verified)

Reply:
they write zeros.

Reply:
It's the same with calldatacopy.

Reply:
On the other hand evmone seems to fill in zeroes apparently...

Reply:
geth code may actually *not* fill in zeroes, though... although I'm not yet sure :-)...

Reply:
For the record: I was just too tired to read the yellowpaper properly, so all fine.

Reply:
What if the address referenced by `extcodecopy` is not valid i.e., there is no code there? Does it still fill in as many zeroes as the `size` parameter dictates?

```
{
  mstore8(2, 100)
  extcodecopy(invalid_address, 0, 0, 32)
  sstore(0, mload(0))
}
```

stores zeroes at storage slot zero post optimization.

Reply:
There is no such thing as an invalid or non-existing address in the EVM. It should still fill with zeros.

Reply:
Are you saying this is turned into `sstore(0, 0)`? The optimizer should not be able to know that the code is empty at that address.

Reply:
No, what I mean is that the optimizer removed the `mstore` which made me think if removal is correct even when `extcodecopy` does not reference a valid address. Good to know that EVM clients should still fill with zeroes in which case the optimizer is correct.